### PR TITLE
Issue #2980: Allow styles to include only a tag without a style.

### LINKED ISSUE #2980: [UX] Allow the use of the <small> tag in ckeditor
## Describe your issue or idea

I often want to make text in various text areas smaller. Unfortunately Backdrop's implementation of ckeditor is lacking the ability to do so :(

From stack overflow: https://stackoverflow.com/questions/24171606/ckeditor-format-tags-and-a-custom-small-tag

 Search for CKEDITOR.config.format_h6 in ckeditor.js and add CKEDITOR.config.format_small={element:"small"};. 

open the languagefile you want (e.g. en.js) and add "tag_small":"small text".
now CKEditor supports the small tag.

### Steps to reproduce (if reporting a bug)

Attempt to wrap some text in the `<small>` tag.

### Actual behavior (if reporting a bug)

There's no `<small>` tag.

### Expected behavior (if reporting a bug)

A `<small>` tag.

### PR DESCRIPTION
**Pull request by quicksketch**
Fixes https://github.com/backdrop/backdrop-issues/issues/2980.

This is only a start, not quite finished yet.

### CODE REVIEW

File: core/modules/ckeditor/ckeditor.admin.inc [hunk 1]
```diff
@@ -76,7 +85,7 @@ function ckeditor_settings_form_validate($form, &$form_state) {
 
   $styles = _ckeditor_settings_parse_style_list($settings['plugins']['style']['style_list']);
   foreach ($styles as $style) {
-    if (empty($style['name']) || empty($style['element']) || empty($style['attributes']['class'])) {
+    if (empty($style['name']) || empty($style['element']))   {
       form_error($form['editor_settings']['plugins']['style']['style_list'], t('The CKEditor list of styles is not formatted properly. Be sure to include one style per line, with the format "element.class|Label".'));
```
Comment (opi) [inline core/modules/ckeditor/ckeditor.admin.inc:89]: Add "or element|Label" into error message ? Same for `#description` on line 76.

File: core/modules/ckeditor/js/ckeditor.admin.js [hunk 2]
```diff
@@ -29,6 +29,7 @@ Backdrop.behaviors.ckeditorAdmin = {
       // Remove the invalid buttons after a delay to allow all behaviors to
       // finish attaching.
       window.setTimeout(function() {
+        JSON.parse($textarea.val());
```
Comment (opi) [inline core/modules/ckeditor/js/ckeditor.admin.js:32]: Is this needed ? Then add a comment about it.

**Review by opi — COMMENTED** [review]
Tested, works great.

**Comment by opi** [comment]
Some minor comment on the PR, but everything looks good and works well.

Also, If you want to include this little UX Sugar please do, otherwise I'll create another PR:
```diff
diff --git a/core/modules/ckeditor/ckeditor.admin.inc b/core/modules/ckeditor/ckeditor.admin.inc
index 64ebf621a..18e416f7d 100644
--- a/core/modules/ckeditor/ckeditor.admin.inc
+++ b/core/modules/ckeditor/ckeditor.admin.inc
@@ -62,7 +62,7 @@ function ckeditor_settings_form(&$form, $form_state, $format) {
   }
   $elements['plugins']['style']['style_list'] = array(
     '#type' => 'textarea',
-    '#rows' => 4,
+    '#rows' => max(4, sizeof(explode("\n", $style_list))),
     '#default_value' => $style_list,
     '#description' => t('A list of classes that will be provided in the "Styles" dropdown. Enter one class on each line in the format: element.class|Label. Example: h1.title|Title.') . '<br />' . t('Each style should be in your theme\'s main CSS as well as in your theme\'s ckeditor-iframe.css file.'),
   );
```

File: core/modules/ckeditor/ckeditor.admin.inc [hunk 3]
```diff
@@ -76,7 +85,7 @@ function ckeditor_settings_form_validate($form, &$form_state) {
 
   $styles = _ckeditor_settings_parse_style_list($settings['plugins']['style']['style_list']);
   foreach ($styles as $style) {
-    if (empty($style['name']) || empty($style['element']) || empty($style['attributes']['class'])) {
+    if (empty($style['name']) || empty($style['element']))   {
```
Comment (klonos) [inline core/modules/ckeditor/ckeditor.admin.inc:88]: Too many spaces there.

**Comment by quicksketch** [comment]
Replaced by https://github.com/backdrop/backdrop/pull/2799.

**Pull request closed without merging**